src/arrayMethodReduce.js

```diff
@@ -5,7 +5,23 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let prev = initialValue;
```

Comment:
```suggestion
    let prev = initialValue !== undefined ? initialValue : this[0];
```

Reply:
in this way the test 'should handle undefined if it was passed as initialValue' doesn't work, so I have done a bit another way. or probably I didn't understand something?